### PR TITLE
Remove-VSTeamAccessControlEntry

### PR DESCRIPTION
**Pull request by Cadacious**
# PR Summary

This pull request adds the Remove-VSTeamAccessControlEntry function. This function removes one or more specified descriptors from an ACL inside the specified security namespace. This function accepts pipeline input for the namespace ID, or a properly constructed VSTeamSecurityNamespace object from which it can extract the namespace ID. Help and Unit tests are included.

## PR Checklist

- [Complete] [Write Help](https://github.com/DarqueWarrior/vsteam/blob/master/.github/CONTRIBUTING.md#write-help)
- [Complete] [Write Unit Test](https://github.com/DarqueWarrior/vsteam/blob/master/.github/CONTRIBUTING.md#write-unit-test)
- [Complete] [Update VSTeam.psd1](https://github.com/DarqueWarrior/vsteam/blob/master/.github/CONTRIBUTING.md#add-a-format-file)
- [Complete] [Update CHANGELOG.md](https://github.com/DarqueWarrior/vsteam/blob/master/.github/CONTRIBUTING.md#add-a-format-file)

### CODE REVIEW

**Comment by DarqueWarrior** [comment]
Thanks for the support. Reviewing the PR now.

File: Source/Public/Remove-VSTeamAccessControlEntry.ps1 [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+       [Parameter(ParameterSetName = 'ByNamespaceId', Mandatory = $true)]
+       [ValidateScript({
+          try {
+              [System.Guid]::Parse($_) | Out-Null
+              $true
+          } catch {
+              $false
+          }
+       })]
+       [string] $SecurityNamespaceId,
```
Comment (DarqueWarrior) [inline Source/Public/Remove-VSTeamAccessControlEntry.ps1]: If you change the type to [guid] you should not need the verification script. I just searched the code base and found other examples where I let this slip through. Change this to [guid] and I will issue PRs to correct those other areas as well.

File: Source/Public/Remove-VSTeamAccessControlEntry.ps1 [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+                catch
+                {
+                    Throw "Could not convert base64 string to string."
+                }
+                $UniqueDescriptor = "Microsoft.TeamFoundation.Identity;"+"$UniqueDescriptor"
+
+                $Descriptors += $UniqueDescriptor
+            }
+
+            [Uri]$Uri = "$($env:TEAM_ACCT)/_apis/accesscontrolentries/$($securityNamespaceId)?token=$($token)&descriptors=$($descriptors -join ",")&api-version=5.1"
```
Comment (DarqueWarrior) [inline Source/Public/Remove-VSTeamAccessControlEntry.ps1]: Don't hard code the API Version. Get rid of

```PowerShell
api-version=5.1
```

File: docs/Remove-VSTeamAccessControlEntry.md [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+Removes specified ACEs in the ACL for the provided token. The request URI contains the namespace ID, the target token, and a single or list of descriptors that should be removed. Only supports removing AzD based users/groups.
+
+## SYNTAX
+
+## DESCRIPTION
+
+Removes specified ACEs in the ACL for the provided token. The request URI contains the namespace ID, the target token, and a single or list of descriptors that should be removed. Only supports removing AzD based users/groups.
+
+## EXAMPLES
+
```
Comment (DarqueWarrior) [inline docs/Remove-VSTeamAccessControlEntry.md]: An example of using this would be great for those that have never seen it before.

Comment (Cadacious) [reply docs/Remove-VSTeamAccessControlEntry.md]: Added examples to help file in latest commit

File: Source/Public/Remove-VSTeamAccessControlEntry.ps1 (hunk 2 above)
Comment (Cadacious) [reply Source/Public/Remove-VSTeamAccessControlEntry.ps1]: Removed hard coded API version in latest commit

File: Source/Public/Remove-VSTeamAccessControlEntry.ps1 (hunk 1 above)
Comment (Cadacious) [reply Source/Public/Remove-VSTeamAccessControlEntry.ps1]: Changed to [GUID] from [String] in latest commit.

**Comment by DarqueWarrior** [comment]
Hi @Cadacious just checking in to see if you saw my last comment about the FromBase64String calls

**Comment by Cadacious** [comment]
I dont see any previous comments about FromBase64String calls.

**Comment by SebastianSchuetze** [comment]
@DarqueWarrior could you comment on what you had to say about @Cadacious FromBase64String. I also don't know what you wanted to address. Otherwise I can have the checks and process the PR.

**Comment by DarqueWarrior** [comment]
Looking into this now.

**Pull request merged**